include/drivers/spi.h

```diff
@@ -176,6 +245,46 @@ struct spi_config {
 	const struct spi_cs_control *cs;
 };
 
+/**
+ * @brief Structure initializer for spi_config from devicetree
+ *
+ * This helper macro expands to a static initializer for a <tt>struct
+ * spi_config</tt> by reading the relevant @p frequency, @p slave, and
+ * @p cs data from the devicetree.
+ *
+ * Important: the @p cs field is initialized using
+ * SPI_CS_CONTROL_PTR_DT(). The @p gpio_dev value pointed to by this
+ * structure must be checked using device_is_ready() before use.
+ *
+ * @param node_id Devicetree node identifier for the SPI device whose
+ *                struct spi_config to create an initializer for
+ * @param operation_ the desired @p operation field in the struct spi_config
+ * @param delay_ the desired @p delay field in the struct spi_config's
+ *               spi_cs_control, if there is one
+ */
+#define SPI_CONFIG_DT(node_id, operation_, delay_)			\
```

Comment:
Should we have a note in `<drivers/spi.h>` mentioning that `SPI_CONFIG_DT` here needs to be maintained in concert with the structure definition in terms of members and order?

Reply:
This file **is** drivers/spi.h, and this macro is being defined directly underneath `struct spi_config`, so I'm hoping that's close enough :wink:

Reply:
Ah; the joys of diffs.  OK.